internal/testutil/fakeclient/tracker.go

```diff
@@ -128,6 +130,16 @@ func (t *TransformingObjectTracker) List(gvr schema.GroupVersionResource, gvk sc
 		return obj, fmt.Errorf("failed to externalize object: %w", err)
 	}
 
+	// Set a fake resource version, so that watches work.
+	if external, ok := external.(interface {
+		GetResourceVersion() string
+		SetResourceVersion(string)
+	}); ok {
+		if external.GetResourceVersion() == "" {
+			external.SetResourceVersion(strconv.FormatInt(time.Now().UnixMilli(), 10))
+		}
+	}
```

Comment:
Slightly confusing shadowing

```suggestion
	if versionedResource, ok := external.(interface {
		GetResourceVersion() string
		SetResourceVersion(string)
	}); ok && versionedResource != nil {
		if versionedResource.GetResourceVersion() == "" {
			versionedResource.SetResourceVersion(strconv.FormatInt(time.Now().UnixMilli(), 10))
		}
	}
```

Added a nil-check for good measure. I guess `external` could theoretically be a `runtime.Object` typed nil pointer 🤔

Reply:
Good point about the nil interface value, that goes under the radar way too quickly. However, I think it's fair not to add the nil check: An `Externalize` implementation shouldn't return such a bogus interface in the first place. I think this would be a programming error in that function, and that it would be fair to panic because of it in a test case. If `Externalize` cannot produce a meaningful externalized object, then it should fail with an error instead. Moreover, maybe the backing methods are fine with a nil receiver, e.g. treat the calls as no-ops.